clock_control: int is returned on enum return value

**Describe the bug**
When using clock_control.h in an app, it does not compile due to warnings.
The return value of the function in an enum, but in case the device is not ready, a posix error code is returned.

```C
static inline enum clock_control_status clock_control_get_status(const struct device *dev,
								 clock_control_subsys_t sys)
{
	const struct clock_control_driver_api *api =
		(const struct clock_control_driver_api *)dev->api;

	if (!api->get_status) {
		return CLOCK_CONTROL_STATUS_UNKNOWN;
	}

	if (!device_is_ready(dev)) {
		return -ENODEV;
	}

	return api->get_status(dev, sys);
}
```
**To Reproduce**
Compile a C++ app that includes clock_control.h

**Expected behavior**
Only return values from the enum

**Impact**
APP does not compile

**Logs and console output**
```
zephyr/include/drivers/clock_control.h:193:11: error: invalid conversion from 'int' to 'clock_control_status' [-fpermissive]
  193 |   return -ENODEV;
      |           ^~~~~~
      |           |
      |           int
```

--- a/include/drivers/clock_control.h
+++ b/include/drivers/clock_control.h
@@ -43,6 +43,7 @@ enum clock_control_status {
 	CLOCK_CONTROL_STATUS_STARTING,
 	CLOCK_CONTROL_STATUS_OFF,
 	CLOCK_CONTROL_STATUS_ON,
+	CLOCK_CONTROL_STATUS_UNAVAILABLE,
 	CLOCK_CONTROL_STATUS_UNKNOWN
 };
 
@@ -190,7 +191,7 @@ static inline enum clock_control_status clock_control_get_status(const struct de
 	}
 
 	if (!device_is_ready(dev)) {
-		return -ENODEV;
+		return CLOCK_CONTROL_STATUS_UNAVAILABLE;
 	}
 
 	return api->get_status(dev, sys);

